scripts/driverkit/build/build.py

```diff
@@ -0,0 +1,151 @@
+        s3_bucket = args.s3_bucket
+        s3_prefix = args.s3_prefix.lstrip('/')
+
+    dri_dirs = [x for x in config_dir.iterdir() if x.is_dir()]
+    for dri_dir in dri_dirs:
+        driverversion = dri_dir.name
+        print(f"[*] loading drivers from driver version directory {driverversion}")
+        files = list(dri_dir.glob("*.yaml"))
+        print(f"[*] found {len(files)} files")
+
+        count = 0
+        success_count = 0
+        fail_count = 0
+        skip_count = 0
+        for config_file in files:
+            count += 1
+            print('[*] [{:03d}/{:03d}] {}'.format(count, len(files), config_file.name))
+
+            with open(config_file) as fp:
+                conf = yaml.safe_load(fp)
+
+            module_output = conf.get('output', {}).get('module')
+            probe_output = conf.get('output', {}).get('probe')
+            
+            module_s3key = None
+            if module_output is not None:
+                module_basename = os.path.basename(module_output)
+                module_s3key = f"{s3_prefix}/{driverversion}/{module_basename}"
+
+            probe_s3key = None
+            if probe_output is not None:
+                probe_basename = os.path.basename(probe_output)
+                probe_s3key = f"{s3_prefix}/{driverversion}/{probe_basename}"
+
+            if s3:
+                need_module = (module_output is not None) and not s3_exists(s3, s3_bucket, module_s3key)
+                need_probe = (probe_output is not None) and not s3_exists(s3, s3_bucket, probe_s3key)
+            else:
+                need_module = (module_output is not None) and not os.path.exists(module_output)
+                need_probe = (probe_output is not None) and not os.path.exists(probe_output)
+
+            need_build = need_module or need_probe
+            if not need_build:
+                skip_count += 1
+                print('[*] {} already built'.format(config_file))
+                continue
+
+            # Make sure the output directory exists or driverkit will output "open: no such file or directory"
+            if module_output is not None:
+                Path(module_output).parent.mkdir(parents=True, exist_ok=True)
+            if probe_output is not None:
+                Path(probe_output).parent.mkdir(parents=True, exist_ok=True)
+
+            success = driverkit_build(driverkit, config_file, driverversion, args.moduledevicename, args.moduledrivername)
+            if success:
+                print(f"[+] Build completed {config_file}")
+                success_count += 1
+            else:
+                print(f"[-] Build failed {config_file}")
+                fail_count += 1
+                continue
+
+            # upload to s3 and remove
+            if s3:
+                if module_output is not None:
+                    with open(module_output, 'rb') as fp:
+                        s3.upload_fileobj(fp, s3_bucket, module_s3key) # XXX TODO ACL for public read
```

Comment:
note for the reviewer: I'm pretty sure I need to set ACLs correctly to allow for public read, but I currently only tested this on a private S3 bucket so I'm not sure how to do it right.

Reply:
added that.